src/foam/mlang/mlangJava.js

```diff
@@ -441,24 +441,67 @@ return ( s1 != null && s2 != null && s1.toUpperCase().contains(s2.toUpperCase())
 
 
 foam.CLASS({
+  refines: 'foam.mlang.predicate.StartsWith',
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+Object arg1 = getArg1().f(obj);
+String arg2 = (String) getArg2().f(obj);
+
+if ( arg1 instanceof Object[] ) {
+  Object[] values = (Object[]) arg1;
+  for ( int i = 0 ; i < values.length ; i++ ) {
+    if ( values[i] instanceof String && ((String) values[i]).startsWith(arg2) )
+      return true;
+  }
+} else if ( arg1 instanceof String ) {
+  return ((String) arg1).startsWith(arg2);
+}
+return false;`
+    },
+    {
+      name: 'createStatement',
+      javaReturns: 'String',
+      args: [{
+        name: 'table',
+        javaType : 'String'
+      }],
+      javaCode: 'return "";'
```

Comment:
The SQL for starts with is:

LIKE 'startString%'

Reply:
I don't really know how these functions work to be honest.